pkg/storaged/btrfs/subvolume.jsx

```diff
@@ -162,6 +166,93 @@ function subvolume_create(volume, subvol, parent_dir) {
     });
 }
 
+function subvolume_delete(volume, subvol, mount_point_in_parent) {
+    const block = client.blocks[volume.path];
+    const subvols = client.uuids_btrfs_subvols[volume.data.uuid];
+
+    function get_direct_subvol_children(subvol) {
+        function is_direct_parent(sv) {
+            return (sv.pathname.length > subvol.pathname.length &&
+                        sv.pathname.substring(0, subvol.pathname.length) == subvol.pathname &&
+                        sv.pathname[subvol.pathname.length] == "/" &&
+                        sv.pathname.substring(subvol.pathname.length + 1).indexOf("/") == -1);
+        }
+
+        return subvols.filter(is_direct_parent);
+    }
+
+    function get_subvol_children(subvol) {
+        // The deepest nested children must come first
+        const direct_children = get_direct_subvol_children(subvol);
+        return flatten(direct_children.map(get_subvol_children)).concat(direct_children);
+    }
+
+    function remove_configs() {
+        return for_each_async(configs_to_remove, c => block.RemoveConfigurationItem(c, {}));
+    }
+
+    function delete_all_subvols() {
+        return cockpit.spawn(["btrfs", "subvolume", "delete"].concat(paths_to_delete),
+                             { superuser: true, err: "message" });
+    }
```

Comment:
We can simplify this if everything is 4.18:

```
                Since 4.18, the ``rmdir(2)`` syscall can delete an empty subvolume just like an
                ordinary directory. Whether this is possible can be detected at runtime, see
                *rmdir_subvol* feature in *FILESYSTEM FEATURES*.
```

Checking happens via `/sys/fs/btrfs/features/rmdir_subvol` seems a worthy check to do in CI and it allows us to get rid of `delete_all_subvols` and `paths_to_delete` although we still need to iterate over it.

Reply:
This already looks pretty simple to me, tbh.

Reply:
Sure, we can keep it, just an alternative approach :)